python/ray/data/_internal/pipelined_dataset_iterator.py

```diff
@@ -26,7 +26,9 @@ def __repr__(self) -> str:
     def _get_next_dataset(self) -> "Dataset":
         if self._epoch_iterator is None:
             self._epoch_iterator = self._base_dataset_pipeline.iter_epochs()
-        ds = next(self._epoch_iterator)
+
+        # The epoch iterator will always have 1 dataset.
```

Comment:
Is this necessarily true? If we're streaming, then we call `dataset.window(bytes_per_window=stream_window_size).repeat()`. In this case, wouldn't there be multiple datasets in an epoch?

```
>>> dataset = ray.data.range(100000)
>>> pipeline = dataset.window(blocks_per_window=1).repeat()
>>> len(list(epoch.iter_datasets()))                                                             
20
```

Reply:
where are you calling `iter_epochs` here?

Reply:
updated to not use `iter_dataset`